src/tainting/Taint.mli

```diff
@@ -16,8 +16,9 @@ type 'a call_trace =
 type sink = { pm : Pattern_match.t; rule_sink : Rule.taint_sink }
 [@@deriving show]
 
-type arg_pos = string * int [@@deriving show]
-type arg = { pos : arg_pos; offset : IL.name list } [@@deriving show]
+type arg_pos = { name : string; index : int } [@@deriving show]
+type arg_base = Bthis | Barg of arg_pos [@@deriving show]
+type arg = { base : arg_base; offset : IL.name list } [@@deriving show]
```

Comment:
This would benefit from some comments and concrete example.
The whole module could also have a toplevel comment explaining the whole approach (and for example how this approach makes our tainting context-sensitive in Semgrep-pro because of our use of signatures).

Reply:
For example I don't really understand what offset is here. An example would really help.

Reply:
Sure thing, I'll revisit the existing comments and add some more.